### PR TITLE
Add “More control over `:nth-child()` selections with the `of S` syntax” article

### PR DESCRIPTION
**Pull request by bramus**
Already filing this PR so editorial stuff can happen.

PR Marked as draft because:

- https://github.com/mdn/browser-compat-data/pull/18852 needs to get merged so BrowserCompat block is up-to-date
- The included demos need to be migrated to the web-dot-dev CodePen account

### CODE REVIEW

**Comment by bramus** [comment]
Ready for review now. Not to be merged until new release of `mdn/browser-compat-data` drops _(will follow-up on this)_

File: site/en/articles/css-nth-child-of-s/index.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+title: More control over `:nth-child()` selections with the `of S` syntax
+authors:
+  - bramus
+subhead: >
+  Pre-filter a set of child elements before applying `An+B` logic on it.
+description: >
+  Pre-filter a set of child elements before applying `An+B` logic on it.
+date: 2023-02-09
+hero: image/AeNB0cHNDkYPUYzDuv8gInYA9rY2/d3BZ8gLh8WkVPRfJf6mz.jpg
+alt: "Rows of green chairs in the Munich Olympic Stadium"
```
Comment (rachelandrew) [inline site/en/articles/css-nth-child-of-s/index.md:12]:
```suggestion
alt: "Rows of green chairs in the Munich Olympic Stadium."
```

File: site/en/articles/css-nth-child-of-s/index.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+With [the `:nth-child()` pseudo-class selector](https://web.dev/learn/css/pseudo-classes/#nth-child-and-nth-of-type) it is possible to select Elements in the DOM by their index. Using the [`An+B` microsyntax](https://www.w3.org/TR/css-syntax-3/#anb-microsyntax) you get fine control over which elements you want to select.
+
+- `:nth-child(2)`: Select the 2nd child.
+- `:nth-child(2n)`: Select all even children _(2nd, 4th, 6th, 8th, and so on)_.
+- `:nth-child(2n+1)`: Select all odd children _(1st, 3rd, 5th, 7th, and so on)_.
+- `:nth-child(5n+1)`: Select the 1st _(=(5×0)+1)_, 5th _(=(5×1)+1)_, 10th _(=(5×2)+1)_, … child.
+- `:nth-child(5n+2)`: Select the 2nd _(=(5×0)+2)_, 6th _(=(5×1)+2)_, 11th _(=(5×2)+2)_, … child.
+
+{% Aside %}
+To interactively see how the `An+B` logic affects selections use [the `:nth-child` tester](https://css-tricks.com/examples/nth-child-tester/)
```
Comment (rachelandrew) [inline site/en/articles/css-nth-child-of-s/index.md:30]:
```suggestion
To interactively see how the `An+B` logic affects selections use [the `:nth-child` tester](https://css-tricks.com/examples/nth-child-tester/).
```

File: site/en/articles/css-nth-child-of-s/index.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+{% endAside %}
+
+But, there’s more creative selections you can do, if you omit the `A` parameter. For example:
+
+- `:nth-child(n+3)`: Select every child from the 3rd one up _(3rd, 4th, 5th, and so on)_
+- `:nth-child(-n+5)`: Select every child up to the 5th one _(1st, 2nd, 3rd, 4th, 5th)_
```
Comment (rachelandrew) [inline site/en/articles/css-nth-child-of-s/index.md:36]:
```suggestion
- `:nth-child(n+3)`: Select every child from the 3rd one up _(3rd, 4th, 5th, and so on)_.
- `:nth-child(-n+5)`: Select every child up to the 5th one _(1st, 2nd, 3rd, 4th, 5th)_.
```

File: site/en/articles/css-nth-child-of-s/index.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+New in [CSS Selectors Level 4](https://www.w3.org/TR/selectors-4/) is the ability to optionally pass a selector list into `:nth-child()` and `:nth-last-child()`.
+
+```css
+:nth-child(An+B [of S]?)
+:nth-last-child(An+B [of S]?)
+```
+
+When `of S` is specified, the `An+B` logic is only applied onto elements that match the given selector list `S`. This essentially means that you can prefilter children before `An+B` does its thing.
+
+{% BrowserCompat 'css.selectors.nth-child.of_syntax' %}
```
Comment (rachelandrew) [inline site/en/articles/css-nth-child-of-s/index.md:59]:
```suggestion
```

**Review by rachelandrew — APPROVED** [review]
Mostly just adding some fullstops plus removing the BCD thing until the update is merged.

File: site/en/articles/css-nth-child-of-s/index.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
+---
+layout: 'layouts/blog-post.njk'
+title: More control over `:nth-child()` selections with the `of S` syntax
```
Comment (rachelandrew) [inline site/en/articles/css-nth-child-of-s/index.md:3]:
```suggestion
title: More control over :nth-child() selections with the of S syntax
```

**Pull request merged**